lighthouse-core/lib/asset-saver.js

```diff
@@ -229,6 +232,156 @@ function* traceJsonGenerator(traceData) {
+      traceEvents.push(...createFakeNetwork(networkNode.record, timing));
+    }
+  }
+
+  traceEvents.push(createFakeEvaluateScript({childEvents: []}, {
+    startTime: lastEventEndTime + 1000,
+    endTime: lastEventEndTime + 1001,
+  }));
+
+  return {traceEvents};
+
+  /**
+   * @return {LH.TraceEvent}
+   */
+  function createFakeTracingStarted() {
+    const data = {
+      frameTreeNodeId: 1,
+      sessionId: '1.1',
+      page: frame,
+      persistentIds: true,
+      frames: [{frame, url: 'about:blank', name: '', processId: 1}],
+    };
+
+    return {
+      ...baseEvent,
+      ts: baseTs - 1e5,
+      ph: 'I',
+      s: 't',
+      cat: 'disabled-by-default-devtools.timeline',
+      name: 'TracingStartedInPage',
+      args: {data},
+      dur: 0,
+    };
+  }
+
+  /**
+   * @param {{childEvents: LH.TraceEvent[]}} cpuNode
+   * @param {{startTime: number, endTime: number}} timing
+   * @return {LH.TraceEvent}
+   */
+  function createFakeEvaluateScript(cpuNode, timing) {
+    const realEvaluateScriptEvent = cpuNode.childEvents.find(e => e.name === 'EvaluateScript' &&
+      !!e.args.data && !!e.args.data.url);
+    // @ts-ignore
+    const scriptUrl = realEvaluateScriptEvent && realEvaluateScriptEvent.args.data.url;
+    const data = {
+      url: scriptUrl || '',
+      frame,
+      lineNumber: 0,
+      columnNumber: 0,
+    };
+
+    return {
+      ...baseEvent,
+      ph: 'X',
+      name: 'EvaluateScript',
+      ts: ts(timing.startTime),
+      dur: (timing.endTime - timing.startTime) * 1000,
+      args: {data},
+    };
+  }
+
+  /**
+   * @param {LH.WebInspector.NetworkRequest} record
+   * @param {LH.Gatherer.Simulation.NodeTiming} timing
+   * @return {LH.TraceEvent[]}
+   */
+  function createFakeNetwork(record, timing) {
+    requestId++;
+
+    // 0ms requests get super-messed up rendering, make it look like they took 20ms
+    let {startTime, endTime} = timing; // eslint-disable-line prefer-const
+    if (startTime === endTime) endTime += 20;
+
+    const requestData = {requestId: requestId.toString(), frame};
+    /** @type {Omit<LH.TraceEvent, 'name'|'ts'|'args'>} */
+    const baseRequestEvent = {...baseEvent, ph: 'I', s: 't', dur: 0};
+
+    const sendRequestData = {
+      ...requestData,
+      requestMethod: record.requestMethod,
+      url: record.url,
+      priority: record.priority(),
+    };
+
+    const receiveResponseData = {
+      ...requestData,
+      statusCode: record.statusCode,
+      mimeType: record._mimeType,
+      encodedDataLength: record._transferSize,
+      fromCache: false,
+      fromServiceWorker: false,
+    };
+
+    const resourceFinishData = {
+      ...requestData,
+    };
+
+    return [
+      {
+        ...baseRequestEvent,
+        name: 'ResourceSendRequest',
+        ts: ts(startTime),
+        args: {data: sendRequestData},
+      },
+      {
+        ...baseRequestEvent,
+        name: 'ResourceReceiveResponse',
```

Comment:
i had to add this only when `(!record.canceled && !record.failed)`. IIRC it's not present on failed requests.

Reply:
done